File: elixir/two-fer/lib/two_fer.ex
defmodule TwoFer do
  @spec two_fer(String.t()) :: String.t()
  def two_fer(name \\ "you")
  def two_fer(name) when is_binary(name), do: "One for #{name}, one for me"
  def two_fer(_), do: raise FunctionClauseError
end

File: c/two-fer/src/two_fer.c
#include "string.h"
#include "two_fer.h"

void two_fer(char* response, char* name)
{
  response[0] = '\0';

  strcat(response, "One for ");

  if (name == NULL) strcat(response, "you");
  else strcat(response, name);

  strcat(response, ", one for me.");
}
